cpp/src/arrow/flight/sql/example/sqlite_server.cc

```diff
@@ -237,23 +226,82 @@ int32_t GetSqlTypeFromTypeName(const char* sqlite_type) {
 }
 
 class SQLiteFlightSqlServer::Impl {
+ private:
   sqlite3* db_;
-  std::map<std::string, std::shared_ptr<SqliteStatement>> prepared_statements_;
+  std::string db_uri_;
+  std::mutex mutex_;
+  std::unordered_map<std::string, std::shared_ptr<SqliteStatement>> prepared_statements_;
```

Comment:
I see that accesses to `prepared_statements_` are never mutex-protected, is it right?